src/vignette/caching/edge/fastly.clj

```diff
@@ -0,0 +1,47 @@
+(ns vignette.caching.edge.fastly
+  (:require [clj-http.client :as client]
+            [vignette.protocols :refer :all]
+            [wikia.common.logger :as log]
+            [wikia.common.perfmonitoring.core :as perf])
+  (:use [environ.core]))
+
+(def default-fastly-api-url "https://api.fastly.com")
+
+(def fastly-creds {:id       (env :fastly-api-id)
+                   :auth-key (env :fastly-api-auth-key)
+                   :api-url  (env :fastly-api-url default-fastly-api-url)})
+
+(declare purge-request-url
+         api-params)
+
+(defrecord FastlyAPI [id auth-key fastly-api-url]
+  CachePurgeAPI
+  (purge [this uri surrogate-key]
+    (perf/publish {:purge-count 1})
+    (let [response (client/post (purge-request-url (:fastly-api-url this)
+                                                   (:id this)
+                                                   surrogate-key)
+                                (api-params (:auth-key this)))]
+      (if (not= (:status response) 200)
+        (do (perf/publish {:failed-purge-count 1})
+            (log/warn (:body response) {:path uri :key surrogate-key})
```

Comment:
should this be a literal string that we can search for, and maybe put `:body` in the context?